include/uart.h

```diff
@@ -29,41 +29,217 @@ extern "C" {
+ *    If no second buffer was specified receiving is stopped and
+ *    UART_RX_DISABLED event is generated. After that whole process can be
+ *    repeated.
+ *
+ * Any time during reception UART_RX_STOPPED event can occur. It will be
+ * followed by UART_RX_BUF_RELEASED event for every buffer currently passed to
+ * driver and finally by UART_RX_DISABLED event.
+ *
+ * Receiving can be disabled using @ref rx_disable, after calling that function
+ * any data received will be lost, UART_RX_BUF_RELEASED event will be generated
+ * for every buffer currently passed to driver and UART_RX_DISABLED event will
+ * occur.
+ *
+ * Transmitting:
+ * 1. Transmitting starts by @ref tx function.
+ * 2. If whole buffer was transmitted UART_TX_DONE is generated.
+ *    If timeout occurred UART_TX_ABORTED will be generated.
+ *
+ * Transmitting can be aborted using @ref tx_abort, after calling that function
+ * UART_TX_ABORTED event will be generated.
+ *
  */
-#define UART_OPTION_AFCE 0x01
+enum uart_event_type {
+	/** @brief Whole TX buffer was transmitted. */
+	UART_TX_DONE,
+	/** @brief Transmitting aborted due to timeout or @ref tx_abort call.
+	 *
+	 * When flow control is enabled, there is a possibility that TX transfer
+	 * won't finish in the allotted time. Some data may have been
+	 * transferred, information about it can be found in event data.
+	 */
+	UART_TX_ABORTED,
+	/**
+	 * @brief Received data is ready for processing.
+	 *
+	 * This event is generated in two cases:
+	 * - When RX timeout occurred, and data was stored in provided buffer.
+	 *   This can happen multiple times in the same buffer.
+	 * - When provided buffer is full.
+	 */
+	UART_RX_RDY,
+	/**
+	 * @brief Driver requests next buffer for continuous reception.
+	 *
+	 * This event is triggered when receiving has started for a new buffer,
+	 * i.e. it's time to provide a next buffer for a seamless switchover to
+	 * it. For continuous reliable receiving, user should provide another RX
+	 * buffer in response to this event, using uart_rx_buf_rsp() function.
+	 *
+	 * If uart_rx_buf_rsp() function is not called, receiving will stop
+	 * after filling current buffer.
+	 */
+	UART_RX_BUF_REQUEST,
+	/**
+	 * @brief Buffer is no longer used by UART driver.
+	 *
+	 * This event informs that receiving stopped to a buffer and it is no
+	 * longer in usage by UART driver.
+	 */
+	UART_RX_BUF_RELEASED,
+	/**
+	 * @brief RX has been disabled and can be reenabled.
+	 *
+	 * This event is generated whenever receiver has been stopped, disabled
+	 * or finished its operation and can be enabled again using @ref rx_en()
+	 */
+	UART_RX_DISABLED,
+	/**
+	 * @brief RX has stopped due to external event.
+	 *
+	 * Reason is one of uart_rx_stop_reason.
+	 */
+	UART_RX_STOPPED,
+};
 
-/** Common line controls for UART.*/
-#define LINE_CTRL_BAUD_RATE	(1 << 0)
-#define LINE_CTRL_RTS		(1 << 1)
-#define LINE_CTRL_DTR		(1 << 2)
-#define LINE_CTRL_DCD		(1 << 3)
-#define LINE_CTRL_DSR		(1 << 4)
```

Comment:
@pfalcon : Do you know why in the past there was no UART_LINE_CTRL_CTS here?
Is it because device is not driving this line or?